src/main/java/seedu/address/logic/PlanInfoCalculator.java

```diff
@@ -0,0 +1,175 @@
+     *
+     * @return Fat intake to adhere to
+     */
+    public double getFats() {
+        return this.fats;
+    }
+
+    /**
+     * Getter for proteins to be consumed for plan
+     *
+     * @return Protein intake to adhere to
+     */
+    public double getProteins() {
+        return this.proteins;
+    }
+
+    /**
+     * Updates macronutrients class variables according to its percentage in diet plan
+     * and the total amount of calories
+     *
+     * @param macroNutrientComposition Macronutrient percentages
+     * @param calories Calories to calculate percentages against
+     */
+    private void calculateMacros(MacroNutrientComposition macroNutrientComposition,
+                                 double calories) {
+        double carboPercent = macroNutrientComposition.getCarbohydrates();
+        double fatPercent = macroNutrientComposition.getFats();
+        double proteinPercent = macroNutrientComposition.getProteins();
+
+        double carboCalories = (carboPercent / 100.00) * calories;
+        this.carbohydrates = carboCalories / 4.0;
+
+        double fatCalories = (fatPercent / 100.00) * calories;
+        this.fats = fatCalories / 9.0;
+
+        double proteinCalories = (proteinPercent / 100.00) * calories;
+        this.proteins = proteinCalories / 4.0;
+    }
+
+    /**
+     * Calculates the amount of calories needed based on diet plan type
+     *
+     * @param user User information
+     * @param planType Type of diet plan
+     * @return Double value containing appropriate amount of calories
+     */
+    private double calculateCalories(User user, PlanType planType) {
+        Age userAge = user.getAge();
+        int age = userAge.getAge();
+
+        Bmi bmi = user.getBmi();
+        double height = bmi.getHeight();
+        double weight = bmi.getWeight();
+
+        double maintenanceCalories = calculateMaintenance(user);
+
+        switch (planType) {
+        case WEIGHTGAIN:
+            return calculateWeightGain(maintenanceCalories);
+        case WEIGHTLOSS:
+            return calculateWeightLoss(maintenanceCalories);
+        default:
+            return maintenanceCalories;
+        }
+
+    }
+
+    /**
+     * Calculates the amount of calories needed for weight maintenance
+     *
+     * @param user User information
+     * @return Double value containing appropriate amount of calories
+     */
+    private double calculateMaintenance(User user) {
+        Age userAge = user.getAge();
+        int age = userAge.getAge();
+
+        Bmi bmi = user.getBmi();
+        double height = bmi.getHeight();
+        double weight = bmi.getWeight();
+
+        Gender gender = user.getGender();
+
+        double maintenanceCalories = 0;
+        if (gender.equals("M")) {
+            maintenanceCalories = (10.0 * weight) + (6.25 * height) - (5.0 * age) + 5.0;
+        } else {
+            maintenanceCalories = (10.0 * weight) + (6.25 * height) - (5.0 * age) - 161.0;
```

Comment:
Perhaps the magic numbers can be replaced with named constants to make the formula easier to understand :)

Reply:
I see! Thanks for the suggestion!